js/bruh-dash.js

```diff
@@ -7,134 +8,299 @@ var global = window || GLOBAL;
+      }
+      newArr.push(newInnerArr);
+    }
+    return newArr;
   },
 
   // creates an array of elements into groups of length of specified size
-  chunk: function(){
-
+  chunk: function(arr, size){
+    let newArr = [];
+    if(size === 0) {
+      return newArr;
+    } else if(arr.length === 0) {
+      return newArr;
+    } else if(size === arr.length || size > arr.length) {
+      newArr.push(arr);
+      return newArr;
+    } else {
+      while(arr.length > 0){
+        newArr.push(arr.splice(0, size));
+      }
+    }
+    return newArr;
   },
 
   // iterates over elements of a collection and invokes iteratee for each element
   // Note: this should work for arrays and objects
-  forEach: function() {
-
+  forEach: function(clct, funct) {
+    if(Array.isArray(clct) === true) {
+      for(i = 0; i < clct.length; i++) {
+        funct(clct[i]);
+      }
+      return funct;
+    } else if(typeof clct === 'object') {
+      for(let key in clct) {
+        funct(clct[key]);
+      }
+    }
   },
 
   // creates an array of values by running each element in collection thru the iteratee
   // Note: this should work for arrays and objects
-  map: function() {
-
+  map: function(clct, funct) {
+    let arr = [];
+    if(Array.isArray(clct) === true) {
+      for(i = 0; i < clct.length; i++) {
+        arr.push(funct(clct[i]));
+      }
+      return arr;
+    } else if(typeof clct === 'object') {
+      for(let key in clct) {
+        arr.push(funct(clct[key]));
+      }
+      return arr;
+    }
   },
 
-  /*************************
-   *  SUPER STRETCH GOALS!  *
-   *************************/ 
+  /*********************************
+   *  SUPER SAIYAN STRETCH GOALS!  *
+   *********************************/
 
   // iterates over elements of a collection and returns all elements that the predicate returns truthy for
   // Note: this should work for arrays and objects
-  filter: function() {
-
+  filter: function(clct, funct) {
+    let arr = [];
+    if(Array.isArray(clct) === true) {
+      for(i = 0; i < clct.length; i++) {
+        if(funct(clct[i]) == true) {
+          arr.push(clct[i]);
+        }
+      }
+      return arr;
+    } else if(typeof clct === 'object') {
+      let emptyObj = {};
+      for(let [key] in clct) {
+        if(funct(clct[key]) == true) {
+          emptyObj[key] = clct[key];
+        }
+      }
+      return emptyObj;
+    }
   },
 
   // Reduces the collection to a value which is the accumulated result of running each element
   // in the collection through an iteratee
   // Note: this should work for arrays and objects
-  reduce: function() {
-    
+  reduce: function(clct, funct) {
+    let total = 0;
+    if(Array.isArray(clct) === true) {
+      for(i = 0; i < clct.length; i++) {
+        total += clct[i];
+        }
+        return total;
+      } else if(typeof clct === 'object') {
+      for(let [key] in clct) {
+        total += clct[key];
+      }
+      return total;
+    }
```

Comment:
don't accumulate in this function with `total += ...`
instead, invoke the given `funct` for each element in `clct`